streams-contrib/streams-provider-twitter/src/main/java/org/apache/streams/twitter/provider/TwitterFollowingProviderTask.java

```diff
@@ -144,6 +165,57 @@ else if( endpoint.equals("friends") )
         } while (curser != 0 && keepTrying < 10);
     }
 
+    private void collectIds(Long id) {
+        int keepTrying = 0;
+
+        long curser = -1;
+
+        do
+        {
+            try
+            {
+                twitter4j.IDs ids = null;
+                if( endpoint.equals("followers") )
+                    ids = client.friendsFollowers().getFollowersIDs(id.longValue(), curser, max_per_page);
+                else if( endpoint.equals("friends") )
+                    ids = client.friendsFollowers().getFriendsIDs(id.longValue(), curser, max_per_page);
+
+                Preconditions.checkNotNull(ids);
+                Preconditions.checkArgument(ids.getIDs().length > 0);
+
+                for (long otherId : ids.getIDs()) {
+
+                    try {
+                        Follow follow;
+                        if( endpoint.equals("followers") ) {
+                            follow = new Follow()
+                                    .withFollowee(new User().withId(id))
+                                    .withFollower(new User().withId(otherId));
+                        } else if( endpoint.equals("friends") ) {
+                            follow = new Follow()
+                                    .withFollowee(new User().withId(otherId))
+                                    .withFollower(new User().withId(id));
+                        } else {
+                            throw new Exception("endpoint must be set to 'friends' or 'followers'");
```

Comment:
the precondition was already being checked, just outside the method where the compiler couldn't trace.  changed to a not null precondition here.